Sources/Containerization/LinuxPod.swift

```
@@ -0,0 +1,761 @@
+    }
+
+    private static func createDefaultRuntimeSpec(_ containerID: String, podID: String) -> Spec {
+        .init(
+            process: .init(),
+            hostname: containerID,
+            root: .init(
+                path: Self.guestRootfsPath(containerID),
+                readonly: false
+            ),
+            linux: .init(
+                resources: .init(),
+                cgroupsPath: "/container/pod/\(podID)/\(containerID)"
+            )
+        )
+    }
+
+    private func generateRuntimeSpec(containerID: String, config: ContainerConfiguration) -> Spec {
+        var spec = Self.createDefaultRuntimeSpec(containerID, podID: self.id)
+
+        // Process configuration
+        spec.process = config.process.toOCI()
+
+        // General toggles
+        spec.hostname = config.hostname
+
+        // Linux toggles
+        spec.linux?.sysctl = config.sysctl
+
+        // Resource limits (if specified)
+        if let cpus = config.cpus {
+            spec.linux?.resources?.cpu = LinuxCPU(
+                quota: Int64(cpus * 100_000),
+                period: 100_000
+            )
+        }
+        if let memoryInBytes = config.memoryInBytes {
+            spec.linux?.resources?.memory = LinuxMemory(
+                limit: Int64(memoryInBytes)
+            )
+        }
+
+        return spec
+    }
+
+    private static func guestRootfsPath(_ containerID: String) -> String {
+        "/run/container/\(containerID)/rootfs"
+    }
+}
+
+extension LinuxPod {
+    /// Number of CPU cores allocated to the pod's VM.
+    public var cpus: Int {
+        config.cpus
+    }
+
+    /// Amount of memory in bytes allocated for the pod's VM.
+    public var memoryInBytes: UInt64 {
+        config.memoryInBytes
+    }
+
+    /// Network interfaces of the pod.
+    public var interfaces: [any Interface] {
+        config.interfaces
+    }
+
+    /// Add a container to the pod. This must be called before `create()`.
```

nit: maybe worth enforcing the documented behavior (`guard case .initialized = state.phase`) to fail early if called after `create()`, rather than relying on callers to follow the doc comment